### PR TITLE
Optimize reporting

### PR DESCRIPTION
**Pull request by sdreher**

### CODE REVIEW

File: quizblock/models.py [hunk 1]
```diff
@@ -473,24 +470,25 @@ def metadata(self):
...
-            else:  # single choice
-                for a in self._answer_cache:
-                    if a.value == r.first().value:
-                        value = a.id
-                        break
-
-        return smart_str(value)
+        value = None
+        submission = Submission.objects.filter(
+            quiz=self.question.quiz, user=user).order_by("-submitted").first()
```
Comment (nikolas) [inline quizblock/models.py]: I'm just curious, is there a reason you decided not to use `self._submission_cache` here?

Comment (sdreher) [inline quizblock/models.py]: I ran some timings and found it wasn't making much of a difference and was adding to memory overhead.

Comment (nikolas) [inline quizblock/models.py]: I see, maybe it's worth removing that instance variable completely, then.

Comment (sdreher) [inline quizblock/models.py]: I thought I did? What am I missing?

Comment (nikolas) [inline quizblock/models.py]: Oh, you did! Nevermind.. sorry, I got confused.

**Pull request merged**